### PR TITLE
[SPARK-14211][SQL] Remove ANTLR3 based parser

### PR DESCRIPTION
**Pull request by hvanhovell**
### What changes were proposed in this pull request?

This PR removes the ANTLR3 based parser, and moves the new ANTLR4 based parser into the `org.apache.spark.sql.catalyst.parser package`.
### How was this patch tested?

Existing unit tests.

cc @rxin @andrewor14 @yhuai

### CODE REVIEW

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 [hunk 1]
```diff
@@ -54,7 +54,7 @@ statement
         (TBLPROPERTIES tablePropertyList)?
         (AS? query)?                                                   #createTable
     | ANALYZE TABLE tableIdentifier partitionSpec? COMPUTE STATISTICS
-        (identifier | FOR COLUMNS identifierSeq?)                      #analyze
+        (identifier | FOR COLUMNS identifierSeq?)?                     #analyze
```
Comment (andrewor14) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: not related right?

**Comment by andrewor14** [comment]
LGTM

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 (hunk 1 above)
Comment (hvanhovell) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: I had to update the `StatisticsSuite` and change its parser to the `HiveSqlParser` & this error popped up. I could also do this in a separate PR.

Comment (andrewor14) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: it's fine

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala [hunk 2]
```diff
@@ -1,749 +0,0 @@
...
-    "TOK_SHOW_COMPACTIONS" -> "SHOW COMPACTIONS",
-    "TOK_SHOW_CREATETABLE" -> "SHOW CREATE TABLE",
-    "TOK_SHOW_GRANT" -> "SHOW GRANT",
-    "TOK_SHOW_ROLE_GRANT" -> "SHOW ROLE GRANT",
-    "TOK_SHOW_ROLE_PRINCIPALS" -> "SHOW PRINCIPALS",
-    "TOK_SHOW_ROLES" -> "SHOW ROLES",
-    "TOK_SHOW_SET_ROLE" -> "SHOW CURRENT ROLES / SET ROLE",
-    "TOK_SHOW_TRANSACTIONS" -> "SHOW TRANSACTIONS",
-    "TOK_SHOWINDEXES" -> "SHOW INDEXES",
-    "TOK_SHOWLOCKS" -> "SHOW LOCKS")
```
Comment (andrewor14) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala]: actually, what happens now if I run one of these? What exception will we get?

Comment (hvanhovell) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala]: It will produce the same error message, with the same nice message. See: https://github.com/hvanhovell/spark/blob/SPARK-14211/sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveSqlParser.scala#L84-L93

File: dev/deps/spark-deps-hadoop-2.2 [hunk 3]
```diff
@@ -2,7 +2,8 @@ JavaEWAH-0.3.2.jar
 RoaringBitmap-0.5.11.jar
 ST4-4.0.4.jar
 activation-1.1.jar
-antlr-runtime-3.5.2.jar
+antlr-2.7.7.jar
```
Comment (rxin) [inline dev/deps/spark-deps-hadoop-2.2]: 2.7.7 and 3.4?

**Comment by hvanhovell** [comment]
retest this please

**Comment by rxin** [comment]
Thanks - merging in master.
